pandas/core/groupby/indexing.py

```diff
@@ -0,0 +1,184 @@
+from __future__ import annotations
+
+from typing import (
+    Iterable,
+    cast,
+)
+
+import numpy as np
+
+from pandas._typing import (
+    FrameOrSeries,
+    PositionalIndexer,
+)
+from pandas.util._decorators import (
+    cache_readonly,
+    doc,
+)
+
+from pandas.core.dtypes.common import (
+    is_integer,
+    is_list_like,
+)
+
+from pandas.core.groupby import groupby
+from pandas.core.indexes.api import CategoricalIndex
+
+
+class GroupByIndexingMixin:
+    """
+    Mixin for adding .rows to GroupBy.
+    """
+
+    @property
+    def rows(self) -> _rowsGroupByIndexer:
+        return _rowsGroupByIndexer(cast(groupby.GroupBy, self))
+
+
+@doc(GroupByIndexingMixin.rows)
+class _rowsGroupByIndexer:
+    def __init__(self, grouped: groupby.GroupBy):
+        self.grouped = grouped
+
+    def __getitem__(self, arg: PositionalIndexer | tuple) -> FrameOrSeries:
+        with groupby.group_selection_context(self.grouped):
+            if isinstance(arg, tuple):
+                if all(is_integer(i) for i in arg):
+                    mask = self._handle_list(arg)
+
```

Comment:
this is a weird extra line

Reply:
![image](https://user-images.githubusercontent.com/8927258/133128909-06a7f0df-e660-44cb-899e-e3c93afaffa5.png)
Do you want me to get rid of _all_ the blank lines in if-elif-else?